checkov/secrets/plugins/detector_utils.py

```diff
@@ -0,0 +1,252 @@
+}
+
+REGEX_VALUE_SECRET_BY_FILETYPE = {
+    FileType.YAML: YML_PAIR_VALUE_SECRET_REGEX_TO_GROUP,
+    FileType.JSON: JSON_PAIR_VALUE_SECRET_REGEX_TO_GROUP,
+    FileType.TERRAFORM: TERRAFORM_PAIR_VALUE_SECRET_REGEX_TO_GROUP,
+}
+
+SINGLE_LINE_PARSER = {
+    FileType.TERRAFORM: terraform_single_line_parser,
+}
+
+MULTILINE_PARSERS = {
+    FileType.YAML: (
+        (FileType.YAML, yml_multiline_parser),
+    ),
+    FileType.JSON: (
+        (FileType.JSON, json_multiline_parser),
+    ),
+    FileType.TERRAFORM: (
+        (FileType.TERRAFORM, terraform_multiline_parser),
+        (FileType.JSON, json_multiline_parser),
+        (FileType.YAML, yml_multiline_parser),
+    ),
+}
+
+
+def remove_fp_secrets_in_keys(detected_secrets: set[PotentialSecret], line: str) -> None:
+    for detected_secret in detected_secrets:
+        if detected_secret.secret_value and line.replace('"', '').replace("'", '').startswith(
+                detected_secret.secret_value):
+            # Found keyword prefix as potential secret
+            detected_secrets.remove(detected_secret)
+            break
+
+
+def format_reducing_noise_secret(string: str) -> str:
+    return json.dumps(string)
+
+
+def extract_from_string(pattern: dict[Pattern[str], int] | None, string: str) -> set[str]:
+    matches: set[str] = set()
+    if not pattern:
+        return matches
+    for value_regex, group_number in pattern.items():
+        match = value_regex.search(string)
+        if match:
+            matches |= {match.group(group_number)}
+    return matches
+
+
+def detect_secret(
+        scanners: tuple[HighEntropyStringsPlugin, ...],
+        filename: str,
+        line: str,
+        line_number: int = 0,
+        **kwargs: Any,
+) -> set[PotentialSecret]:
+    for entropy_scanner in scanners:
+        matches = entropy_scanner.analyze_line(filename, line, line_number, **kwargs)
+        if matches:
+            return matches
+    return set()
+
+
+def analyze_multiline(
+        filename: str,
+        scanners: tuple[HighEntropyStringsPlugin, ...],
+        multiline_parser: BaseMultiLineParser,
+        line_number: int = 0,
+        context: CodeSnippet | None = None,
+        raw_context: CodeSnippet | None = None,
+        value_pattern: dict[Pattern[str], int] | None = None,
+        secret_pattern: dict[Pattern[str], int] | None = None,
+        **kwargs: Any,
+) -> set[PotentialSecret]:
+    secrets: set[PotentialSecret] = set()
+    if context is None or raw_context is None:
+        return secrets
+    value_secrets = extract_from_string(pattern=secret_pattern, string=context.target_line)
+    for possible_secret in value_secrets:
+        secret_adjust = format_reducing_noise_secret(possible_secret)
+
+        entropy_on_value = detect_secret(
+            scanners=scanners,
+            filename=filename,
+            line=secret_adjust,
+            line_number=line_number,
+            kwargs=kwargs
+        )
+
+        if entropy_on_value:
```

Comment:
it's no more just entropy, but a `potential_secret` which may be detected by scanner other than entropy scanner